src/system-update-generator/system-update-generator.c

```diff
@@ -47,9 +48,41 @@ static int generate_symlink(void) {
         if (symlink(SYSTEM_DATA_UNIT_PATH "/system-update.target", p) < 0)
                 return log_error_errno(errno, "Failed to create symlink %s: %m", p);
 
+        return 1;
+}
+
+static int parse_proc_cmdline_item(const char *key, const char *value, void *data) {
+        assert(key);
+
+        if (streq(key, "systemd.unit")) {
+                if (proc_cmdline_value_missing(key, value))
+                        return 0;
+
+                log_warning("Offline system update overriden by kernel command line systemd.unit= setting");
+
+        } else if (!value) {
+                const char *target;
+
+                target = runlevel_to_target(key);
+                if (target)
+                        log_warning("Offline system update overriden by runlevel \"%s\" on the kernel command line", key);
```

Comment:
No need to save `target` here, just `if (!value && runlevel_to_target(key) != NULL) { warn }' would be more succinct.